src/data_aggregator.ts

```diff
@@ -107,100 +133,120 @@ export class DataAggregator {
 
     // Record any failures
-    for (let i = 0; i < exchangeAdaptersArr.length; i++) {
-      if (allTickerData[i].status === PromiseStatus.REJECTED) {
-        const exchange = exchangeAdaptersArr[i].exchangeName
+    for (let i = 0; i < allPrices.length; i++) {
+      if (allPrices[i].status === PromiseStatus.REJECTED) {
+        const source = this.priceSources[i]
         this.logger.warn(
           {
-            exchange,
-            err: allTickerData[i].value,
+            source,
+            err: allPrices[i].value,
           },
-          'Fetching ticker failed'
+          'Fetching price failed'
+        )
+        this.config.metricCollector?.error(source.name())
+      } else if (allPrices[i].status === PromiseStatus.RESOLVED) {
+        // Record the price using MetricCollector.
+        this.config.metricCollector?.priceSource(
+          this.config.currencyPair,
+          this.priceSources[i].name(),
+          allPrices[i].value
         )
-        this.config.metricCollector?.error(exchange)
       }
     }
 
-    // Remove failed requests
-    const successfulTickerData: Ticker[] = allTickerData
-      .filter((settledTicker: SettledPromise) => settledTicker.status === PromiseStatus.RESOLVED)
-      .map((settledTicker: SettledPromise) => {
-        const ticker: Ticker = settledTicker.value
-        // Record the ticker with the metric collector
-        this.config.metricCollector?.ticker(ticker)
-        return ticker
-      })
+    // Remove failed requests.
+    const successfulPriceData: WeightedPrice[] = allPrices
+      .filter((promise: SettledPromise) => promise.status === PromiseStatus.RESOLVED)
+      .map((promise: SettledPromise) => promise.value)
 
-    assert(successfulTickerData.length > 0, `All ticker requests failed`)
+    assert(successfulPriceData.length > 0, `All price requests failed`)
 
-    return successfulTickerData
+    return successfulPriceData
   }
 
   async currentPrice(): Promise<BigNumber> {
     switch (this.config.aggregationMethod) {
       case AggregationMethod.MIDPRICES:
-        return this.currentPriceFromTickerData()
+        return this.weightedMeanMidPrice()
       default:
         throw Error(`Aggregation method ${this.config.aggregationMethod} not recognized`)
     }
   }
 
-  async currentPriceFromTickerData(): Promise<BigNumber> {
-    const allTickerData = await this.fetchAllTickers()
-    let validTickerData = aggregators.checkIndividualTickerData(
-      allTickerData,
-      this.config.maxPercentageBidAskSpread,
-      this.config.metricCollector,
-      this.logger
-    )
-    validTickerData = aggregators.crossCheckTickerData(validTickerData, this.config)
-    return aggregators.weightedMeanMidPrice(validTickerData)
+  async weightedMeanMidPrice(): Promise<BigNumber> {
+    const allPriceData = await this.fetchAllPrices()
+    let validPriceData = allPriceData
+    validPriceData = aggregators.crossCheckPriceData(validPriceData, this.config)
```

Comment:
Could these 2 lines be simplified to this?

```typescript
    let validPriceData = aggregators.crossCheckPriceData(allPriceData, this.config)
```

Reply:
Indeed. Done.